src/types.py

```diff
@@ -476,14 +491,19 @@ def int64() -> "Type":
         return Type(TypeData(kind=TypeData.K_INT, size=64))
 
     @staticmethod
-    def of_size(size: int) -> "Type":
-        return Type(TypeData(kind=TypeData.K_ANY, size=size))
+    def int_of_size(size: int) -> "Type":
+        return Type(TypeData(kind=TypeData.K_INT, size=size))
```

Comment:
Preventing this from being a `K_CTYPE` is more limiting than the old code (but I think that's OK).

The byte/short/word could be a union, a single-field struct, or a struct with bitfields. So I think this line is responsible for the following diff in OOT's `func_800DB2C0`:
```diff
@@ -14,17 +14,7 @@

     if (temp_v0 < (s32) gAudioContext.gAudioBufferParameters.updatesPerFrame) {
 loop_1:
         temp_v1 = &gAudioContext.gNoteSubsEu[(gAudioContext.gMaxSimultaneousNotes * phi_v0) + arg1];
-        if ((((s32) temp_v1->bitField0.s * 2) < 0) || (temp_v0_2 = phi_v0 + 1, temp_v1->bitField0.s = (struct 
-{
-  vu8 enabled : 1;
-  u8 needsInit : 1;
-  u8 finished : 1;
-  u8 envMixerNeedsInit : 1;
-  u8 stereoStrongRight : 1;
-  u8 stereoStrongLeft : 1;
-  u8 stereoHeadsetEffects : 1;
-  u8 usesHeadsetPanEffects : 1;
-}) ((u8) temp_v1->bitField0.s & 0xFF7F), phi_v0 = temp_v0_2, phi_return = temp_v0_2, phi_return = temp_v0_2, ((temp_v0_2 < (s32) gAudioContext.gAudioBufferParameters.updatesPerFrame) == 0))) {
+        if ((((s32) temp_v1->bitField0.s * 2) < 0) || (temp_v0_2 = phi_v0 + 1, temp_v1->bitField0.s = (u8) temp_v1->bitField0.s & 0xFF7F, phi_v0 = temp_v0_2, phi_return = temp_v0_2, phi_return = temp_v0_2, ((temp_v0_2 < (s32) gAudioContext.gAudioBufferParameters.updatesPerFrame) == 0))) {
 
         } else {
             goto loop_1;
```

*But*, I don't think this is actually a bad regression. I think your change is bringing this constructor to be in line with the rest of the constructors. In the future, we can add deliberate support for small CTypes.

Reply:
That's some pretty funny output from earlier. Yeah, I agree that we should make this as an intentional change later on. It's similar to how we don't support stack structs right now.